### PR TITLE
Initial commit of Ladybug_shadows

### PR DESCRIPTION
**Pull request by Vyrwn**

### CODE REVIEW

**Comment by chriswmackey** [comment]
Good job @Vyrwn !

**Comment by mostaphaRoudsari** [comment]
Looks like a very useful component! You need to update the dates though.
They're  2015!

**Comment by Vyrwn** [comment]
Hi Mostapha,

Thank you very much. Yes I will spend some time during the week to fix most of the problems I can. I had a very useful work-through from Chris yesterday. I will keep you updated.

**Comment by Vyrwn** [comment]
Hi Chris, Mostapha,

I just found 3 hours of peace to work on this while waiting in a coffee shop. I realized how long it has been since the last time.

I updated the component and now it runs with no issues (at least I have not come across any yet). It is done with more function calling than before since I am a bit more proficient with Python now.

As you anticipated Chris, I forgot how to do the whole pull request, clone thingy. I am not certain weather I need to request a copy of Ladybug again and then upload my component or the other way around.

Whenever you have time to have a look let me know.

**Comment by Vyrwn** [comment]
https://www.dropbox.com/s/e7efp7u0f9qs8iu/Ladybug_Visualize_Shadows.ghuser?dl=0

**Comment by mostaphaRoudsari** [comment]
@Vyrwn thank you for the update. You need to sync your fork with this repository and then make your changes and send us a pull request with both the userobject and the .py file. Let us know if you have any difficulties.

**Comment by mostaphaRoudsari** [comment]
@Vyrwn also a number of minor comments about the component.

1. A number of inputs are named incorrectly. As you can see [here](https://github.com/mostaphaRoudsari/ladybug/blob/master/resources/Ladybug%20Grahpic%20Standards.md#hard-and-fast-rules) they should be camelCase.

2. Default hours are missing from the component. It crashes the component. See below.

image

**Comment by Vyrwn** [comment]
Thank you for your comments, hopefully tomorrow I will have some time to do the appropriate changes. Last time I said that it took me just half a year..

**Comment by chriswmackey** [comment]
@Vyrwn ,

Glad to see that you are improving the component and I had meant to share some thoughts from a number of people in my office who have been using your work (it has been a blast with our visualization/rendering team, btw).

First and foremost, many of us think that it might be a good idea to separate your component into two components: one that renders the Rhino scene and one that sets the sun position.  There are a lot of cases where we want to create animations with renderings but do not want to include a Rhino sun.  Having a separate render component might also enable us to expose more settings for rendering (much like the Ladybug Capture View components has).  Let me know your thoughts on this.  I would also be happy to help flesh out a render component.

That's the biggest suggestion.  Once we get your thoughts on that, I can share some of the smaller ones.

-Chris

**Comment by Vyrwn** [comment]
Hi Chris,

At some point yesterday I was wondering whether the component has too many variables making it less flexible. You are correct.

I might take some time to set it up again since I am slightly more experienced now than when I started it. I will take out the render part into a separate component. Introducing more setting for render though is depending from the render engine and I am not the best in that section. I use to do my renders in 3ds-Max, therefore my V-Ray knowledge is limited to a simple ambient occlusion render. 

I can make the core of the component and then you could suggest any additions.

I would love to hear the rest of your thoughts on this.

-Byron

**Comment by chriswmackey** [comment]
@Vyrwn ,

I apologize for the very late response.  I hope you don't mind that the 3D Viz team at my office really needed a render component and so I took out the relevant parts of your code and expanded it to create a "Render Views" component (much like the "Capture Views" component):
https://github.com/mostaphaRoudsari/ladybug/commit/bfc4d1f3e1c304ab831eaca6c69e364018a694ce

I was able to get the component to be fully compatible with both Rhino render and V-Ray render (by tapping into the very small V-Ray API).  In the future, we could add support for more rendering plugins but I think that this should suffice for now.

In summary, if you could send a new Pull Request that has your component just as a component to set the sun position and characteristics, I will merge it in and send you feedback specifically related to the sun position capabilities.

Thanks, as always,
-Chris
